Mathlib/AlgebraicTopology/SimplicialObject/Coskeletal.lean

```diff
@@ -0,0 +1,115 @@
+variable (X : SimplicialObject C) (n : ℕ)
+
+namespace Truncated
+
+/-- The identity natural transformation exhibits a simplicial set as a right extension of its
+restriction along `(Truncated.inclusion (n := n)).op`.-/
+@[simps!]
+def rightExtensionInclusion :
+    RightExtension (Truncated.inclusion (n := n)).op
+      (Truncated.inclusion.op ⋙ X) := RightExtension.mk _ (𝟙 _)
+
+end Truncated
+
+open Truncated
+
+/-- A simplicial object `X` is `n`-coskeletal when it is the right Kan extension of its restriction
+along `Truncated.inclusion (n := n).op` via the identity natural transformation.-/
+class IsCoskeletal : Prop where
+  nonempty_isRightKanExtension :
+    Nonempty (IsRightKanExtension X (𝟙 (Truncated.inclusion (n := n).op ⋙ X)))
+
+variable [X.IsCoskeletal n]
+
+/-- If `X` is `n`-cosketal, then `rightExtensionInclusion X n` is a terminal object in the category
+`RightExtension (Truncated.inclusion (n := n)).op (Truncated.inclusion.op ⋙ X)`. -/
+theorem IsCoskeletal.isRightKanExtension :
+    IsRightKanExtension X (𝟙 (Truncated.inclusion (n := n).op ⋙ X)) :=
+  IsCoskeletal.nonempty_isRightKanExtension.some
+
+/-- If `X` is `n`-cosketal, then `rightExtensionInclusion X n` is a terminal object in the category
+`RightExtension (Truncated.inclusion (n := n)).op (Truncated.inclusion.op ⋙ X)`. -/
+noncomputable def IsCoskeletal.isUniversalOfIsRightKanExtension :
+    (rightExtensionInclusion X n).IsUniversal := by
+  have := isRightKanExtension X n
+  apply Functor.isUniversalOfIsRightKanExtension
+
+variable [∀ (F : (SimplexCategory.Truncated n)ᵒᵖ ⥤ C),
+    SimplexCategory.Truncated.inclusion.op.HasRightKanExtension F]
+
+/-- There is a map of costructured arrows from `rightExtensionInclusion X n` to the right extension
+of the `n`-truncation of `X` defined by the counit of `coskAdj n`.-/
+noncomputable def Truncated.coskRightExtension.hom : Truncated.rightExtensionInclusion X n ⟶
+    RightExtension.mk _
+      ((coskAdj n).counit.app ((Truncated.inclusion (n := n)).op ⋙ X)) :=
+  CostructuredArrow.homMk ((coskAdj n).unit.app X) ((coskAdj n).left_triangle_components X)
+
+instance Truncated.coskIsRightKanExtension : IsRightKanExtension
+    ((Truncated.cosk n).obj (inclusion.op ⋙ X)) ((coskAdj n).counit.app (inclusion.op ⋙ X)) := by
+  unfold Truncated.cosk
+  unfold coskAdj
+  rw [ranAdjunction_counit]
+  exact Functor.instIsRightKanExtensionObjRanAppRanCounit _ _
+
+/-- The map `coskRightExtension.hom X` is a natural transformation between two right Kan extensions
+of the diagram `Truncated.inclusion.op ⋙ X` and thus is an isomorphism. -/
+instance IsCoskeletal.coskRightExtension.hom_isIso : IsIso (coskRightExtension.hom X n) :=
+  isIso_of_isTerminal (IsCoskeletal.isUniversalOfIsRightKanExtension X n)
+    (((Truncated.cosk n).obj
+      ((Truncated.inclusion (n := n)).op ⋙ X)).isUniversalOfIsRightKanExtension
+        ((coskAdj n).counit.app ((Truncated.inclusion (n := n)).op ⋙ X)))
+      (coskRightExtension.hom X n)
```

Comment:
I'm a bit confused by how the naming conventions are being used here. I think usually this first instance would be called `isRightKanExtensionCosk`? But I see the tension with putting `cosk` first.

Reply:
Similarly for `isIsoHom`?